src/IndexManagement/IndexManager.php

```diff
@@ -213,19 +213,27 @@ function (array $indexProperties): array {
     /**
      * @inheritDoc
      */
-    public function reindex(string $source, string $destination): void
-    {
-        $this->execute(
-            function () use ($source, $destination) {
+    public function reindex(
+        string $source,
+        string $destination,
+        bool $waitForCompletion = true,
+        bool $proceedConflicts = false,
+        bool $reindexJustMissingDocuments = false
+    ): array {
```

Comment:
looks a bit much already... what do you think about introducing an `$options` array instead?